nx-X11/programs/Xserver/hw/nxagent/NXdixfonts.c

```diff
@@ -936,19 +939,43 @@ SetDefaultFontPath(char *path)
                 size = 0,
                 bad;
 
+    /* ensure temp_path contains "built-ins" */
+    start = path;
+    while (1) {
+	start = strstr(start, "built-ins");
+	if (start == NULL)
+	    break;
+	end = start + strlen("built-ins");
+	if ((start == path || start[-1] == ',') && (!*end || *end == ','))
+	    break;
+	start = end;
+    }
+    if (!start) {
+	if (asprintf(&temp_path, "%s%sbuilt-ins", path, *path ? "," : "")
+	    == -1)
+	    temp_path = NULL;
+	}
+    else {
+	temp_path = strdup(path);
+    }
+    if (!temp_path)
+	return BadAlloc;
+
     /* get enough for string, plus values -- use up commas */
 #ifdef NX_TRANS_SOCKET
-    len = strlen(_NXGetFontPath(path)) + 1;
+    len = strlen(_NXGetFontPath(temp_path)) + 1;
```

Comment:
> This will override temp_path if the NX_FONT env variable is set. Maybe we need logic like that 
> in there (_NXGetFontPath()) as well to always have the built-ins string appended if needed?

I think I found a better solution for this, I put...

```
start = _NXGetFontPath(path)
```

at the beginning of the function (instead of start = path).